SSR-2824 User guide for ReadOnlyTable (#1211)
kintone-labs/kintone-ui-component: docs/document/docs/components/desktop/readonly-table.md

| Name   | Type | Default | Description | Remark |
| :--- | :--- | :--- | :--- | :--- |

> | className | string | ""  | Component class name. | |

Reviewer: ```suggestion
| className | string | ""  | Component class name | |
```

> | id | string | ""  | Component id name. | |

Reviewer: ```suggestion
| id | string | ""  | Component id name | |
```

| label | string | ""  | Text description for the component. | Label will not be displayed if unspecified or left empty. |
| rowsPerPage | number | 5 | Number of table rows per page. | <ul><li>Round off to the nearest whole number. </li><ul> <li> `{rowsPerPage: 10}` Display up to 10 rows per page. </li><li> `{rowsPerPage: 20.5}` Display up to 21 rows per page. </li> <li> `{rowsPerPage: 20.4}` Display up to 20 rows per page. </li></ul><li>Throw an error if `rowsPerPage` is not a positive number in constructor and setter.</li><ul>|
| pagination | boolean | true | Show/Hide the pagination. | <ul><li>`{pagination: false}` Pagination is hidden, all rows are displayed.</li><li>`{pagination: true}` Only the number of rows set in rowsPerPage are displayed.</li></ul> |
| visible | boolean | true | Show/Hide the component. | |

> | columns | Array\<[Columns](#column)\> | []  | Column data of the component. | Throw an error if the `columns` is not an array. |

Reviewer: ```suggestion
| columns | Array\<[Column](#column)\> | []  | Column data of the component. | Will result an error if the value of `columns` is not an array. |
```

> | data | Array\<object\> | []  | Row data of the component. | Throw an error if the `data` is not an array. |

Reviewer: ```suggestion
| data | Array\<object\> | []  | Row data of the component | Will result an error if the value of `data` is not an array. |
```

> #### Column

Reviewer: How about writing inside "### Property" like this?
Column.field, Column.title, etc.
https://kintone-ui-component.netlify.app/docs/en/components/desktop/dropdown

| Name | Type | Default | Description | Remark |
| :--- | :--- | :--- | :--- | :--- |

> | field | string | ""  | Key of the data object. | |

Reviewer: ```suggestion
| field | string | ""  | Key of the data object | |
```

> | title | string | ""  | Header name of the column. | |

Reviewer: ```suggestion
| title | string | ""  | Header name of the column | |
```

> | visible | boolean |  true  | Show/Hide the column. | |

Reviewer: ```suggestion
| visible | boolean |  true  | Show/Hide the column | |
```

### Constructor

> 

Reviewer: ```suggestion

> Please check the [package installation](../../getting-started/quick-start.md#installation) method first.
```

| id | string | ""  | Component id name | |
| label | string | ""  | Label for the component | Label will not be displayed if unspecified or left empty |
| visible | boolean | true | Show/Hide the component | |
| columns | Array\<Column\> | []  | Array of data to be displayed in the table header | Will result an error if the value for columns is not an array |
| Column.title | String | "" | The header name of each column | |
| Column.field | String | "" | The key of the data object | |

> | Column.visible | Boolean | true | Show/Hide the column | |

Reviewer 2: Separate column property to a table

Here is a sample code when all parameters are specified:

```javascript
const Kuc = Kucs['1.x.x'];

const space = kintone.app.record.getSpaceElement('space');

const readOnlyTable = new Kuc.ReadOnlyTable({
  rowsPerPage: 3,
  pagination: true,
  columns: [
    {
      title: 'Number',
      field: 'index',
    },
    {
      title: 'City',
      field: 'name',
    },
    {
      title: 'Country',
      field: 'country',
    },
    {
      title: 'Population',
      field: 'population',
    },
    {
      title: 'Coordinates',
      field: 'coordinates',
    }
  ],
  data: [
    {
      index: '1',
      name: 'HoChiMinh',
      country: 'Vietnam',
      population: '8,371,000',
      coordinates: '10.762622, 106.660172',
    },
    {
      index: '2',
      name: 'Tokyo',
      country: 'Japan',
      population: '14,000,000',
      coordinates: '35.689487, 139.691711',
    },
    {
      index: '3',
      name: 'New York',
      country: 'USA',
      population: '8,400,000',
      coordinates: '40.712776, -74.005974',
    },
  ],
});
space.appendChild(readOnlyTable);

> ```

Reviewer: Please update with below including every props.

```
const Kuc = Kucs['1.x.x'];

const space = kintone.app.record.getSpaceElement('space');

const readOnlyTable = new ReadOnlyTable({
  label: 'ReadOnlyTable',
  columns: [
    {
      title: 'Number',
      field: 'index',
    },
    {
      title: 'City',
      field: 'name',
    },
    {
      title: 'Country',
      field: 'country',
    },
    {
      title: 'Population',
      field: 'population',
    },
    {
      title: 'Coordinates',
      field: 'coordinates',
    }
  ],
  data: [
    {
      index: '1',
      name: 'HoChiMinh',
      country: 'Vietnam',
      population: '8,371,000',
      coordinates: '10.762622, 106.660172',
    },
    {
      index: '2',
      name: 'Tokyo',
      country: 'Japan',
      population: '14,000,000',
      coordinates: '35.689487, 139.691711',
    },
    {
      index: '3',
      name: 'New York',
      country: 'USA',
      population: '8,400,000',
      coordinates: '40.712776, -74.005974',
    }
  ],
  className: 'sample-class',
  id: 'sample-id',
  visible: true,
  pagination: true,
  rowsPerPage: 3,
});
space.appendChild(readOnlyTable);
```
